zrml/prediction-markets/src/lib.rs

```diff
@@ -1158,6 +1162,97 @@ mod pallet {
             let assets_len: u32 = assets.len().saturated_into();
             Ok(Some(T::WeightInfo::sell_complete_set(assets_len)).into())
         }
+
+        /// When the `MaxDisputes` amount of disputes is reached,
+        /// this allows to start a global dispute.
+        ///
+        /// # Arguments
+        ///
+        /// * `market_id`: The identifier of the market.
+        ///
+        /// NOTE:
+        /// The outcomes of the disputes and the report outcome
+        /// are added to the global dispute voting outcomes.
+        /// The bond of each dispute is the initial vote amount.
+        #[pallet::weight(T::WeightInfo::start_global_dispute(CacheSize::get()))]
+        #[transactional]
+        pub fn start_global_dispute(
+            origin: OriginFor<T>,
+            #[allow(dead_code, unused)]
+            #[pallet::compact]
+            market_id: MarketIdOf<T>,
+        ) -> DispatchResultWithPostInfo {
+            ensure_signed(origin)?;
+
+            #[allow(dead_code, unused)]
+            let mut weight: Weight = T::WeightInfo::start_global_dispute(CacheSize::get());
+
+            #[cfg(feature = "with-global-disputes")]
+            {
+                let market = T::MarketCommons::market(&market_id)?;
+                ensure!(market.status == MarketStatus::Disputed, Error::<T>::InvalidMarketStatus);
+
+                ensure!(
+                    market.dispute_mechanism == MarketDisputeMechanism::SimpleDisputes,
+                    Error::<T>::InvalidDisputeMechanism
+                );
+
+                let disputes = <Disputes<T>>::get(market_id);
+                ensure!(
+                    disputes.len() == T::MaxDisputes::get() as usize,
+                    Error::<T>::MaxDisputesNeeded
+                );
+
+                ensure!(
+                    T::GlobalDisputes::is_not_started(&market_id),
+                    Error::<T>::GlobalDisputeAlreadyStarted
+                );
+
+                // add report outcome to voting choices
+                if let Some(report) = market.report {
+                    let res = T::GlobalDisputes::push_voting_outcome(
+                        &market_id,
+                        report.outcome,
+                        &report.by,
+                        <BalanceOf<T>>::zero(),
+                    )
+                    .map_err(|_| log::warn!("Push of report voting outcome failed."));
+                    debug_assert!(res.is_ok());
+                }
+
+                for (index, MarketDispute { at: _, by, outcome }) in disputes.iter().enumerate() {
+                    let dispute_bond = default_dispute_bond::<T>(index);
+                    let res = T::GlobalDisputes::push_voting_outcome(
+                        &market_id,
+                        outcome.clone(),
+                        by,
+                        dispute_bond,
+                    )
+                    .map_err(|_| log::warn!("Push of dispute voting outcome failed."));
```

Comment:
I can't answer directly on this previous review comment https://github.com/zeitgeistpm/zeitgeist/pull/682#discussion_r1009376656 . So I will take the opportunity here. I can't really find the code reference anymore. For the other benchmarks in the pm pallet I often use `0..63` (benchmarks are actually range inclusive -> so it is 64 times like the CacheSize). It's not that important to do it right in this case, because the weight is averaged out over most of the possible values. But yeah, better to be exact ;-)

Reply:
> Why don't we just error when pushing the voting outcome fails?

Good question. The thought was, that global disputes is the resolution mechanism of last resort. So I wanted to make sure, that there are as few error cases as possible to actually finally get a decision over a dispute. That's why I used `debug_assert` to catch the mistake in the early dev cycle with the tests. The only error for `push_voting_outcome` is the `GlobalDisputeAlreadyFinished` error. 

So in this case I think your suggestion is great. Lets do it. 3c81de5d9db6f2ad540beb16684c02a01a1daabd